synapse_data_warehouse/synapse/dynamic_tables/R__populate_acl_latest_table.sql

```diff
@@ -0,0 +1,47 @@
+use schema {{database_name}}.synapse; --noqa: JJ01,PRS,TMP,CP01
+
+CREATE DYNAMIC TABLE IF NOT EXISTS ACL_LATEST
+    TARGET_LAG = '1 day'
+    WAREHOUSE = compute_xsmall
+    AS
+        WITH acl_expanded AS (
+            SELECT
+                owner_id,
+                change_timestamp,
+                COALESCE(
+                    array_sort(value:"accesstype"::variant),
+                    array_sort(value:"accessType"::variant),
+                    array_sort(value:"accesstype#1"::variant),
+                    array_sort(value:"accesstype#2"::variant),
+                    array_sort(value:"accesstype#3"::variant)
+                ) AS access_type,
+                COALESCE(
+                    value:"principalId"::number,
+                    value:"principalid"::number,
+                    value:"principalid#1"::number
+                ) AS principal_id
+            FROM 
+                {{database_name}}.synapse_raw.aclsnapshots,
+                LATERAL FLATTEN(input => parse_json(resource_access), outer => TRUE)
+        ),
```

Comment:
Hi @philerooski  thanks for the feedback! Agreed, it makes sense to work with temp or transient tables rather than CTEs for clarity. I went with using the temp tables because they're session-specific, but added an extra bit of logic to drop the tables after the dynamic table is created, just in case. Changes are in https://github.com/Sage-Bionetworks/snowflake/pull/74/commits/6b2222d6eacc00dd938da3791bf8afafd260fb72

Let me know if this improves the structure and flow of the script & if you have any other suggestions!